model/src/main/scala/org/broadinstitute/dsde/rawls/model/DataReferenceModel.scala

```diff
@@ -1,35 +1,85 @@
 package org.broadinstitute.dsde.rawls.model
 
-import bio.terra.workspace.model.{DataReferenceDescription, DataReferenceList}
+import java.util.UUID
+
+import bio.terra.workspace.model._
 import org.broadinstitute.dsde.workbench.model.{ValueObject, ValueObjectFormat}
 import spray.json.DefaultJsonProtocol._
+import spray.json.{DeserializationException, JsArray, JsObject, JsString, JsValue, RootJsonFormat, enrichAny}
 
 import scala.collection.JavaConverters._
 
 case class DataReferenceName(value: String) extends ValueObject
-case class DataRepoSnapshot(name: DataReferenceName, snapshotId: String)
-case class DataRepoSnapshotReference(referenceId: String, name: String, workspaceId: String, referenceType: Option[String], reference: Option[String], cloningInstructions: String)
-case class DataRepoSnapshotList(snapshots: List[DataRepoSnapshotReference])
-case class TerraDataRepoSnapshotRequest(instanceName: String, snapshot: String)
-
-object DataRepoSnapshotReference {
-  def apply(ref: DataReferenceDescription): DataRepoSnapshotReference = {
-    DataRepoSnapshotReference(ref.getReferenceId.toString, ref.getName, ref.getWorkspaceId.toString,
-      Option(ref.getReferenceType.toString), Option(ref.getReference), ref.getCloningInstructions.toString)
+case class NamedDataRepoSnapshot(name: DataReferenceName, snapshotId: String)
+
+object DataReferenceModelJsonSupport {
+  implicit object DataRepoSnapshotFormat extends RootJsonFormat[DataRepoSnapshot] {
+    val INSTANCE_NAME = "instanceName"
+    val SNAPSHOT = "snapshot"
+
+    def write(snap: DataRepoSnapshot) = JsObject(
+      INSTANCE_NAME -> JsString(snap.getInstanceName),
+      SNAPSHOT -> JsString(snap.getSnapshot)
+    )
+
+    def read(json: JsValue) = {
+      json.asJsObject.getFields(INSTANCE_NAME, SNAPSHOT) match {
+        case Seq(JsString(instanceName), JsString(snapshot)) =>
+          new DataRepoSnapshot().instanceName(instanceName).snapshot(snapshot)
+        case _ => throw new DeserializationException("DataRepoSnapshot expected")
```

Comment:
nit: in Scala, if you are creating an instance of a case class, you don't need - and it's idiomatic to omit - the `new` keyword:
```
case _ => throw DeserializationException("DataRepoSnapshot expected")
```
( this applies elsewhere in the PR too )